pkg/pool-stable-phantom/contracts/test/MockStablePhantomPool.sol

```diff
@@ -15,14 +15,57 @@
 pragma solidity ^0.7.0;
 pragma experimental ABIEncoderV2;
 
+import "@balancer-labs/v2-pool-utils/contracts/test/MockFailureModes.sol";
+
 import "../StablePhantomPool.sol";
 
-contract MockStablePhantomPool is StablePhantomPool {
+contract MockStablePhantomPool is StablePhantomPool, MockFailureModes {
     constructor(NewPoolParams memory params) StablePhantomPool(params) {
         // solhint-disable-previous-line no-empty-blocks
     }
 
-    function mockCacheTokenRateIfNecessary(IERC20 token) external {
+    function mockCacheTokenRateIfNecessary(IERC20 token) external whenNotInFailureMode(FailureMode.PRICE_RATE) {
         _cacheTokenRateIfNecessary(token);
     }
+
+    function getRate()
+        public
+        view
+        virtual
+        override
+        whenNotInFailureMode(FailureMode.INVARIANT)
+        whenNotInFailureMode(FailureMode.PRICE_RATE)
```

Comment:
I'd remove the price rate modifier from here and add it to `_scalingFactors`. Same with swaps and joins. That'll let us more accurately simulate what actually happens (i.e. that getting the scaling factors fails).